### PR TITLE
chore: MMI adds note to trader support to the new Tx confirmation view

### PR DESCRIPTION
**Pull request by zone-live**
## **Description**

Adds the MMI transaction "note to trader" functionality to the new transactions redesign, the current one is also kept in order to facilitate backwards compatibility with the current confirmation view (to be removed later).
It also adds a few parameters that were missing for MMI like the tx metadata.

note_to_trader_ss

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by albertolive — APPROVED** [review]

File: ui/components/institutional/note-to-trader/note-to-trader-tabbed.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+type NoteToTraderProps = {
+  placeholder: string;
+  maxLength: number;
+  onChange: (value: string) => void;
+  noteText: string;
+  labelText: string;
+};
+
+const TabbedNoteToTrader: React.FC<NoteToTraderProps> = ({
```
Comment (zone-live) [inline ui/components/institutional/note-to-trader/note-to-trader-tabbed.tsx:17]: Will be removed towards the end of the year.

File: ui/components/institutional/note-to-trader/note-to-trader.tsx [hunk 2]
```diff
@@ -33,29 +59,31 @@ const NoteToTrader: React.FC<NoteToTraderProps> = ({
           display={Display.Flex}
           justifyContent={JustifyContent.spaceBetween}
         >
-          <Label htmlFor="transaction-note">{labelText}</Label>
+          <Label htmlFor="transaction-note">{t('transactionNote')}</Label>
           <Text className="note-header__counter">
-            {noteText.length}/{maxLength}
+            {noteText.length}/{280}
```
Comment (pedronfigueiredo) [inline ui/components/institutional/note-to-trader/note-to-trader.tsx:64]: I'm not sure what's the intended behaviour with this, but do we want to do `{noteText.length}/280` or `{noteText.length}/{MAX_LENGTH}` with `const MAX_LENGTH = 280`?

File: ui/components/institutional/note-to-trader/note-to-trader.tsx [hunk 3]
```diff
@@ -1,28 +1,54 @@
...
-    <Box className="confirm-page-container-content__data">
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      dispatch(setNoteToTraderMessage(noteText));
+    }, 700);
+
+    return () => clearTimeout(timer);
+  }, [noteText]);
+
+  return isNoteToTraderSupported && !isSignature ? (
```
Comment (pedronfigueiredo) [inline ui/components/institutional/note-to-trader/note-to-trader.tsx:45]: [nit]: Can instead be expressed with an early return which is arguably more readable:

```tsx
if (!isNotToTraderSupported || isSignature) {
  return null
}
```

File: ui/pages/confirmations/confirm/confirm.tsx [hunk 4]
```diff
@@ -63,6 +64,11 @@ const Confirm = () => (
               <Title />
               <Info />
               <PluggableSection />
+              {
```
Comment (pedronfigueiredo) [inline ui/pages/confirmations/confirm/confirm.tsx:67]: 🔥

**Review by pedronfigueiredo — APPROVED** [review]
LGTM!

File: ui/components/institutional/note-to-trader/note-to-trader.tsx (hunk 2 above)
Comment (zone-live) [reply ui/components/institutional/note-to-trader/note-to-trader.tsx:64]: the 280 is the max length. I'll assign it to a const 👍🏼

**Pull request merged**